weave/integrations/google_ai_studio/google_ai_studio_sdk.py

```diff
@@ -0,0 +1,87 @@
+import importlib
+from functools import wraps
+from typing import TYPE_CHECKING, Any, Callable, Union
+
+if TYPE_CHECKING:
+    from google.generativeai.types.generation_types import GenerateContentResponse
+
+import weave
+from weave.trace.op_extensions.accumulator import add_accumulator
+from weave.trace.patcher import MultiPatcher, SymbolPatcher
+
+
+def gemini_accumulator(
+    acc: Union["GenerateContentResponse", None], value: "GenerateContentResponse"
+) -> "GenerateContentResponse":
+    if acc is None:
+        acc = value
+    if not acc._done:
+        return value
+    if acc != value:
```

Comment:
Is this `if` correct?  What if the acc is done and it's the same as the value (e.g. you return just 1 token)

Reply:
If i don't add this `if`, the first token is being accumulated twice, both in the weave trace as well as in the output of the function. I agree that this is not the best way to tackle this issue and I'm open to suggestions to improve the logic here.

Reply:
Are you saying that adding the accumulator modifies the output of the function?  It shouldn't do that -- we should never modify the user's output

Reply:
Yeah, I placed the check to prevent modification of output.

Reply:
sorry, I don't understand.

Here's what I think I'm reading:
1. If the acc is fresh, use the first value of the iterator
2. While the acc is not finished, use the next values
3. If acc != value, then aggregate the candidate completions

So open questions are:
1. How does the above modify the user's output?  It shouldn't.  We should only be "listening" to the stream
2. Why do we need the acc != value check?  I guess there's an interaction with `_done`, but I'm not sure how

Reply:
Done.